### PR TITLE
fix: props destructuring for functions passed as props

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

Changed all direct use of functions passed as prop to destructured usage. This change was needed because of dev config conflict with design system package.

Fixes [#16685](https://github.com/appsmithorg/appsmith/issues/16685)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested using yalc in local

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/design-system/src/ColorSelector/index.tsx [hunk 1]
```diff
@@ -58,6 +58,7 @@ const ColorBox = styled.div<{ selected: string; color: string }>`
 `;
 
 function ColorSelector(props: ColorSelectorProps) {
+  const { onSelect } = props;
   const [selected, setSelected] = useState<string>(
     props.defaultValue || props.colorPalette[0],
```
Comment (ankitakinger) [inline packages/design-system/src/ColorSelector/index.tsx:63]: defaultValue and colorPalette can also be destructured?

File: packages/design-system/src/ColorPicker/index.tsx [hunk 2]
```diff
@@ -155,11 +156,11 @@ interface ColorPickerProps {
 }
 
 function ColorPickerComponent(props: ColorPickerProps) {
+  const { changeColor } = props;
   const [color, setColor] = React.useState(props.color);
```
Comment (ankitakinger) [inline packages/design-system/src/ColorPicker/index.tsx:160]: color can also be destructured?

File: packages/design-system/src/Callout/index.tsx [hunk 3]
```diff
@@ -93,7 +94,7 @@ function Callout(props: CalloutProps) {
       <Text type={TextType.P2}>{props.text}</Text>
       {props.label ? props.label : null}
       {props.closeButton ? (
-        <Label onClick={props.onClose} variant={props.variant || Variant.info}>
+        <Label onClick={onClose} variant={props.variant || Variant.info}>
```
Comment (ankitakinger) [inline packages/design-system/src/Callout/index.tsx:97]: label & variant can also be destructured?

**Review by ankitakinger — APPROVED** [review]
LGTM

**Review by tanvibhakta — APPROVED** [review]

**Pull request merged**